src/test/java/seedu/address/model/task/TaskListTest.java

```diff
@@ -107,4 +112,33 @@ public void deleteTask_validIndex_success() {
     public void getTaskList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> typicalTaskList.getTaskList().remove(0));
     }
+
+    @Test
+    public void findIndexOfTask_earlierDeadline_success() {
+        int results = timedTaskList.findIndexOfTask(TASK_DEADLINE_BEFORE_CHICKEN);
+        int expectedResults = 1;
+        assertEquals(results, expectedResults);
+    }
+
+    @Test
+    public void findIndexOfTask_afterDeadline_success() {
+        int results = timedTaskList.findIndexOfTask(TASK_DEADLINE_AFTER_CHICKEN);
+        int expectedResults = 3;
+        assertEquals(results, expectedResults);
+    }
+
+    @Test
+    public void findIndexOfTask_sameDeadline_success() {
+        int results = timedTaskList.findIndexOfTask(TASK_DEADLINE_SAME_CHICKEN);
+        int expectedResults = 1;
+        assertEquals(results, expectedResults);
+    }
+
+    @Test
+    public void findIndexOfTask_emptyTaskList_success() {
+        int results = taskList.findIndexOfTask(TASK_DEADLINE_AFTER_CHICKEN);
+        int expectedResults = 0;
+        assertEquals(results, expectedResults);
+    }
```

Comment:
Comprehensive test cases, good to see that test cases check for empty task list as well.